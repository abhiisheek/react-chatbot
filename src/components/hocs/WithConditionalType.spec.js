import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import withConditionalType from './WithConditionalType';

const typesMap = {
  text: 'Text',
  chart: 'Chart'
};

const conditionalFunc = ({ type }) => typesMap[type];

const WrappedComponent = props => <div>{props.children}</div>;

WrappedComponent.displayName = 'Wrapped';

WrappedComponent.propTypes = {
  children: PropTypes.element.isRequired
};

describe('WithConditionalType', () => {
  it('Snapshot test', () => {
    const ConditionalComponent = withConditionalType(WrappedComponent);
    const wrapper = renderer
      .create(
        <ConditionalComponent type="text" conditionalFunc={conditionalFunc} />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Should generate HOC component which renders component on the type passed', () => {
    const ConditionalComponent = withConditionalType(WrappedComponent);

    const textWrapper = shallow(
      <ConditionalComponent type="text" conditionalFunc={conditionalFunc} />
    );

    expect(textWrapper.html()).toBe('<div><Text type="text"></Text></div>');

    const chartWrapper = shallow(
      <ConditionalComponent type="chart" conditionalFunc={conditionalFunc} />
    );

    expect(chartWrapper.html()).toBe('<div><Chart type="chart"></Chart></div>');

    expect(ConditionalComponent.displayName).toBe('WrappedWithConditionalType');
  });

  it('Should generate HOC component by with default name when not set on the component', () => {
    WrappedComponent.displayName = undefined;

    const ConditionalComponent = withConditionalType(WrappedComponent);

    const textWrapper = shallow(
      <ConditionalComponent type="text" conditionalFunc={conditionalFunc} />
    );

    expect(textWrapper.html()).toBe('<div><Text type="text"></Text></div>');

    expect(ConditionalComponent.displayName).toBe(
      'ComponentWithConditionalType'
    );
  });
});
