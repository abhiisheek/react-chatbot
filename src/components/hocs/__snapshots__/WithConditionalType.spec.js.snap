// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithConditionalType Snapshot test 1`] = `
<div>
  <Text
    type="text"
  />
</div>
`;
